lib/Utils/TimberDynamicResize.php

```diff
@@ -0,0 +1,303 @@
+    {
+        $uploadDir = wp_upload_dir();
+        return $uploadDir['baseurl'];
+    }
+
+    public function getUploadsBasedir()
+    {
+        $uploadDir = wp_upload_dir();
+        return $uploadDir['basedir'];
+    }
+
+    public function resizeDynamic(
+        $src,
+        $w,
+        $h = 0,
+        $crop = 'default',
+        $force = false
+    ) {
+        if ($this->enabled) {
+            $resizeOp = new Resize($w, $h, $crop);
+            $fileinfo = pathinfo($src);
+            $resizedUrl = $resizeOp->filename(
+                $fileinfo['dirname'] . '/' . $fileinfo['filename'],
+                $fileinfo['extension']
+            );
+
+            if (empty($this->flyntResizedImages)) {
+                add_action('shutdown', [$this, 'storeResizedUrls'], -1);
+            }
+            $this->flyntResizedImages[$w . '-' . $h . '-' . $crop] = [$w, $h, $crop];
+
+            return $this->addImageSeparatorToUploadUrl($resizedUrl);
+        } else {
+            return $this->generateImage($src, $w, $h, $crop, $force);
+        }
+    }
+
+    public function registerRewriteRule($wpRewrite)
+    {
+        $routeName = static::IMAGE_QUERY_VAR;
+        $relativeUploadDir = $this->getRelativeUploadDir();
+        $relativeUploadDir = trailingslashit($relativeUploadDir) . static::IMAGE_PATH_SEPARATOR;
+        $wpRewrite->rules = array_merge(
+            ["^{$relativeUploadDir}/?(.*?)/?$" => "index.php?{$routeName}=\$matches[1]"],
+            $wpRewrite->rules
+        );
+    }
+
+    public function addRewriteTag()
+    {
+        $routeName = static::IMAGE_QUERY_VAR;
+        add_rewrite_tag("%{$routeName}%", "([^&]+)");
+    }
+
+    public function checkAndGenerateImage($relativePath)
+    {
+        $matched = preg_match('/(.+)-(\d+)x(\d+)-c-(.+)(\..*)$/', $relativePath, $matchedSrc);
+        $exists = false;
+        if ($matched) {
+            $originalRelativePath = $matchedSrc[1] . $matchedSrc[5];
+            $originalPath = trailingslashit($this->getUploadsBasedir()) . $originalRelativePath;
+            $originalUrl = trailingslashit($this->getUploadsBaseurl()) . $originalRelativePath;
+            $exists = file_exists($originalPath);
+            $w = (int) $matchedSrc[2];
+            $h = (int) $matchedSrc[3];
+            $crop = $matchedSrc[4];
+        }
+
+        if ($exists) {
+            global $wpdb;
+            $tableName = $this->getTableName();
+            $resizedImage = $wpdb->get_row(
+                $wpdb->prepare("SELECT * FROM {$tableName} WHERE width = %d AND height = %d AND crop = %s", [
+                    $w,
+                    $h,
+                    $crop,
+                ])
+            );
+        }
+
+        if (empty($resizedImage)) {
+            header("HTTP/1.0 404 Not Found");
+            exit();
+        }
+
+        $resizedUrl = $this->generateImage($originalUrl, $w, $h, $crop);
+
+        header("Location: {$resizedUrl}", true, 302);
```

Comment:
Should we consider using the WP `status_header` function? https://developer.wordpress.org/reference/functions/status_header/